modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -0,0 +1,480 @@
+
+      params.set<MeshGeneratorName>("input") = name() + "trans";
+      params.set<std::vector<Point>>("normals") = {
+          {1, 0, 0},
+          {-1, 0, 0},
+          {0, 1, 0},
+          {0, -1, 0}}; // normal directions over which to define boundaries
+      params.set<bool>("fixed_normal") = true;
+      params.set<bool>("replace") = false;
+      params.set<std::vector<BoundaryName>>("new_boundary") = {"10001", "10002", "10003", "10004"};
+
+      _build_mesh = &addMeshSubgenerator("SideSetsFromNormalsGenerator", name() + "_2D", params);
+    }
+  }
+  else
+  {
+    // Hex geometry
+    {
+      auto params = _app.getFactory().getValidParams("PolygonConcentricCircleMeshGenerator");
+      params.set<unsigned int>("num_sides") = 6; // Cartesian geometry so pin are given a square box
+      params.set<std::vector<unsigned int>>("num_sectors_per_side") =
+          std::vector<unsigned int>(6, _num_sectors);
+      params.set<bool>("preserve_volumes") = true;
+      params.set<bool>("quad_center_elements") = _quad_center;
+      params.set<boundary_id_type>("external_boundary_id") =
+          20000 +
+          _pin_type; // The boundary id for pins as set by the Reactor Geometry Meshing convention
+      params.set<std::string>("external_boundary_name") = "outer_pin_" + std::to_string(_pin_type);
+
+      // Break the mesh intervals and region ids up by construct for
+      // PolygonConcentricCircleMeshGenerator
+      std::vector<unsigned int> ring_intervals;
+      std::vector<subdomain_id_type> ring_ids;
+      unsigned int background_intervals = 1;
+      std::vector<subdomain_id_type> background_ids;
+      std::vector<unsigned int> duct_intervals;
+      std::vector<subdomain_id_type> duct_ids;
+      for (const auto i : make_range(_intervals.size()))
+      {
+        if (i < _ring_radii.size())
+        {
+          ring_intervals.push_back(_intervals[i]);
+          ring_ids.push_back(_region_ids[0][i]);
+        }
+        else if (i > _ring_radii.size())
+        {
+          duct_intervals.push_back(_intervals[i]);
+          duct_ids.push_back(_region_ids[0][i]);
+        }
+        else
+        {
+          background_intervals = _intervals[i];
+          background_ids.push_back(_region_ids[0][i]);
+        }
+      }
+      if (ring_intervals.size() > 0)
+      {
+        params.set<std::vector<Real>>("ring_radii") = _ring_radii;
+        if (ring_intervals.front() != 1)
+          ring_ids.insert(ring_ids.begin(), ring_ids.front());
+
+        params.set<std::vector<subdomain_id_type>>("ring_block_ids") = ring_ids;
+        params.set<std::vector<unsigned int>>("ring_intervals") = ring_intervals;
+      }
+      else if (background_intervals > 1)
+      {
+        background_ids.insert(background_ids.begin(), background_ids.front());
+      }
+
+      params.set<std::vector<subdomain_id_type>>("background_block_ids") = background_ids;
+      params.set<unsigned int>("background_intervals") = background_intervals;
```

Comment:
duplicated from above. Do it before the if statement to consolidate

Reply:
Moved to the beginning of the if block, I left any lines with `params.set` as is just to keep that separate from the defining of the id-related variables